src/k8s_helpers.rs

```diff
@@ -15,16 +24,28 @@ fn create_secret(name: &str, data: BTreeMap<String, ByteString>) -> Secret {
     }
 }
 
-pub fn create_secret_from_files(
+pub fn create_secret(
     secret_name: &str,
-    key_files: &[(PathBuf, &str)], //[pathbuf, key type]
+    secrets: HashMap<String, SecretType>,
 ) -> Result<Secret, Box<dyn Error>> {
-    let mut data = BTreeMap::new();
-    for (file_path, key_type) in key_files {
-        let file_content = std::fs::read(file_path)
-            .map_err(|err| format!("Failed to read file '{:?}': {}", file_path, err))?;
-        data.insert(format!("{}.json", key_type), ByteString(file_content));
+    let mut data: BTreeMap<String, ByteString> = BTreeMap::new();
+    for (label, value) in secrets {
+        match value {
+            SecretType::Value { v } => {
+                data.insert(label, ByteString(v.into_bytes()));
+            }
+            SecretType::File { path } => {
+                let file_content = std::fs::read(&path)
+                    .map_err(|err| format!("Failed to read file '{:?}': {}", path, err))?;
+                data.insert(label, ByteString(file_content));
+            }
+        }
```

Comment:
nit: you might be able to do this in one statement by collecting into a `BTreeMap`, ie:

```
    let data = secrets.into_iter().map(|(label_value)| ... (label, ByteString(v.into_bytes()))).collect::<BTreeMap<_, _>>();
```